### ISSUE TITLE
clearml logs 0 instead of nan

### ISSUE THREAD

**Issue by paantya** [allegroai/clearml#604]
clearml logs 0 instead of NaN in bords.

expect that should be kept nan

**Comment by paantya** [comment]
https://clearml.slack.com/archives/CTK20V944/p1646928807931049

Hello!
tell me please, is it intended that nan values are converted to 0 when logging?
upd: I see NAN in the tensorboard, and 0 in Clearml.
upd2: use v1.1.* (edited) 


[natanM](https://app.slack.com/team/U017S2MD849)  [1 day ago](https://clearml.slack.com/archives/CTK20V944/p1646929048886669?thread_ts=1646928807.931049&cid=CTK20V944)
Hi [@pa antya](https://clearml.slack.com/team/U01RCBAPES1),
What are you logging? Can you provide a small snippet or a screenshot? (edited) 
:hourglass_flowing_sand:
1


[pa antya](https://app.slack.com/team/U01RCBAPES1)  [24 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646930582177199?thread_ts=1646928807.931049&cid=CTK20V944)[](https://files.slack.com/files-pri/TT9ATQXJ5-F03632G3XST/image.png)[](https://files.slack.com/files-pri/TT9ATQXJ5-F03632G3XST/download/image.png)[](https://files.slack.com/files-pri/TT9ATQXJ5-F03632HN7M5/newplot__8_.png)[](https://files.slack.com/files-pri/TT9ATQXJ5-F03632HN7M5/download/newplot__8_.png)[](https://files.slack.com/files-pri/TT9ATQXJ5-F03632JKP0F/image.png)[](https://files.slack.com/files-pri/TT9ATQXJ5-F03632JKP0F/download/image.png)
3 files 
image.png


newplot (8).png


image.png



[pa antya](https://app.slack.com/team/U01RCBAPES1)  [24 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646930683829819?thread_ts=1646928807.931049&cid=CTK20V944)[](https://files.slack.com/files-pri/TT9ATQXJ5-F03632VKJHM/download/test_nan_clearml_vs_tb.ipynb)
u can run it (edited) 
Binary 

test_nan_clearml_vs_tb.ipynb
BinaryDownload Binary



[pa antya](https://app.slack.com/team/U01RCBAPES1)  [24 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646930708123519?thread_ts=1646928807.931049&cid=CTK20V944)
[@natanM](https://clearml.slack.com/team/U017S2MD849) ^

[natanM](https://app.slack.com/team/U017S2MD849)  [24 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646930734284399?thread_ts=1646928807.931049&cid=CTK20V944)
[@pa antya](https://clearml.slack.com/team/U01RCBAPES1), I will take a look soon :slightly_smiling_face: (edited) 
:raised_hands:
1
:+1:
1


[pa antya](https://app.slack.com/team/U01RCBAPES1)  [24 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646930765630119?thread_ts=1646928807.931049&cid=CTK20V944)[](https://files.slack.com/files-pri/TT9ATQXJ5-F036EQ0JTFX/download/test_nan_-__iterations.json)
test_nan -  Iterations.json 
[{"task":"9f6d79d810cb481fad6e34f2e1e03563","name":"epoch_0","x":[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19],"y":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"type":"scatter"},{"task":"9f6d79d810cb481fad6e34f2e1e03563","name":"epoch_1","x":[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19],"y":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"type":"scatter"},{"task":"9f6d79d810cb481fad6e34f2e1e03563","name":"epoch_2","x":[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19],"y":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"type":"scatter"}]




[pa antya](https://app.slack.com/team/U01RCBAPES1)  [24 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646930896306639?thread_ts=1646928807.931049&cid=CTK20V944)
[@natanM](https://clearml.slack.com/team/U017S2MD849)
Will wait)
not nice that this logging is misleading

[natanM](https://app.slack.com/team/U017S2MD849)  [23 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646933805033559?thread_ts=1646928807.931049&cid=CTK20V944)
[@pa antya](https://clearml.slack.com/team/U01RCBAPES1), can you point me to where in the script the reported scalars are?
I think this might be happening because you can't report None for Logger.report_scalar() so the auto logging assigns it some sort of value - 0. What is your use case? If the value of the scalar is None then why log it?

[pa antya](https://app.slack.com/team/U01RCBAPES1)  [23 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646933871311019?thread_ts=1646928807.931049&cid=CTK20V944)
```
class LitMNIST(LightningModule):
...
        self.log('test/test_nan', np.nan, prog_bar=False, logger=True, on_step=True, on_epoch=False)
...
```
(edited)

[pa antya](https://app.slack.com/team/U01RCBAPES1)  [23 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646933937892469?thread_ts=1646928807.931049&cid=CTK20V944)[](https://files.slack.com/files-pri/TT9ATQXJ5-F036J2D2D99/download/untitled.py)
all code
Untitled 
```
class LitMNIST(LightningModule):
    def __init__(self, data_dir=PATH_DATASETS, hidden_size=64, learning_rate=2e-4):
​
        super().__init__()
Click to expand inline (94 lines)
```




[pa antya](https://app.slack.com/team/U01RCBAPES1)  [23 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646934093832119?thread_ts=1646928807.931049&cid=CTK20V944)
[@natanM](https://clearml.slack.com/team/U017S2MD849)
usability of the pytorch_lightning logger
we log the average reward of each action for the RL agent.
If the agent you did this action on the current episode, then his average reward will be nan , not 0. for obvious reasons. And we would like it to be visualized in the same way as in the tensorboard, for informational content.

[pa antya](https://app.slack.com/team/U01RCBAPES1)  [23 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646934186294149?thread_ts=1646928807.931049&cid=CTK20V944)
[@natanM](https://clearml.slack.com/team/U017S2MD849)
*If the agent did not perform a certain action, then its average reward per episode for this action will be nan , not 0

[pa antya](https://app.slack.com/team/U01RCBAPES1)  [23 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646934225300059?thread_ts=1646928807.931049&cid=CTK20V944)
```
import numpy as np
np.nan
```

[Martin.B](https://app.slack.com/team/UT8T0V3NE)  [21 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646939938626259?thread_ts=1646928807.931049&cid=CTK20V944)
[@pa antya](https://clearml.slack.com/team/U01RCBAPES1)
upd: I see NAN in the tensorboard, and 0 in Clearml.
I have to admit, since NaN's are actually skipped in the graph, should we actually log them ?

[pa antya](https://app.slack.com/team/U01RCBAPES1)  [20 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646945203503429?thread_ts=1646928807.931049&cid=CTK20V944)
[@Martin.B](https://clearml.slack.com/team/UT8T0V3NE)
if I had to choose between logging or not logging, I would choose logging
If you choose between logging as 0 or as nan, then I would choose as nan
If you choose between skipping or logging like nan, then here I find it difficult, it seems that it is better to log than skip, but you need to think.
to a greater extent, we are used to the tensorboard, where nan is logged in a special way, and this behavior seems to be natural. (edited) 

[Martin.B](https://app.slack.com/team/UT8T0V3NE)  [19 hours ago](https://clearml.slack.com/archives/CTK20V944/p1646945870668009?thread_ts=1646928807.931049&cid=CTK20V944)
 If you choose between skipping or logging like nan, then here I find it difficult, it seems that it is better to log than skip, but you need to think.
So I "think" the issue is plotly (UI), doesn't like NaN and also elastic (storing the scalar) is not a NaN fan. We need to check if they both agree on the representation, that it should be easy to fix...
Maybe you could open a github issue, so we do not forget?

**Comment by paantya** [comment]
![image (2)](https://user-images.githubusercontent.com/2975813/157907739-ec11b114-0937-4e55-87b8-69886201b465.png)
![newplot (9)](https://user-images.githubusercontent.com/2975813/157907759-3283bfd0-a641-4964-9cbc-73fa58cacaa7.png)
![image (1)](https://user-images.githubusercontent.com/2975813/157907762-2fd661f0-72af-48d0-bd10-520269e49170.png)

**Comment by paantya** [comment]
rename file for run 
to `test_nan_clearml_vs_tb (1).ipynb` 

[test_nan_clearml_vs_tb (1).ipynb.txt](https://github.com/allegroai/clearml/files/8233615/test_nan_clearml_vs_tb.1.ipynb.txt)

**Comment by eugen-ajechiloae-clearml** [comment]
Hi @paantya. Thanks for reporting this! As of now, clearml doesn't support logging 'NaN' values or any other strings and they default to a 0 value (as you mentioned). For now, we have decided to warn the user the first time such a value is encountered and allow the user to change the default value. This should be available in a future release.

**Comment by ajecc** [comment]
@paantya 0 to another float when reporting the value to the ClearML platform

**Comment by jkhenning** [comment]
@paantya I think the meaning is to allow the user to specify a value other than the default `0` to be used instead of `NaN` when it is reported

**Comment by ajecc** [comment]
> 

Yes, that's what it means

**Comment by eugen-ajechiloae-clearml** [comment]
> @paantya I think the meaning is to allow the user to specify a value other than the default `0` to be used instead of `NaN` when it is reported

Yes, that's what it means

**Comment by erezalg** [comment]
Hi @paantya, a new  RC with a fix for this is ready
pip install clearml==1.3.2rc1
you can see the new methods as such:
from clearml import Logger

Logger.set_reporting_nan_value()
Logger.set_reporting_inf_value()

**Comment by jkhenning** [comment]
Hi @paantya, closing this. Please re-open if required.
